### ISSUE TITLE
Is this repo for a newer edition of the book?

### ISSUE THREAD

**Issue by atc0005** [bitfield/tpg-tools2#1]
Hi @bitfield,

Is this repo intended to replace the https://github.com/bitfield/tpg-tools repo, or is this one for a follow-up edition of the book?

**Comment by bitfield** [comment]
It's the latter, @atc0005, but as you can see, the programs involved are pretty much the same; I'm just updating the code to the way I would write it now. The next edition of the book will reference this repo instead of the original.

**Comment by atc0005** [comment]
@bitfield thanks for confirming!

Will purchasers of the current edition receive a copy of the new edition? 

I know you did that with your For the Love of Go title, but wasn't sure if that was a blanket policy of yours.

If it is a blanket policy, that is a perk well worth emphasizing. :)

**Comment by bitfield** [comment]
Yes, that's the case for all my books (try re-downloading any of those you've bought previously—there might be updates).

> If it is a blanket policy, that is a perk well worth emphasizing. :)

Good point, I'll add a mention to the product pages.

**Comment by cassamajor** [comment]
@bitfield, not to rush, but I am curious when this new edition will be released.

---
> Separate question I had, and found the answer to. Leaving it here for others.

Q: In what order would you recommend someone read your books? [Are you a Go black belt?](https://bitfieldconsulting.com/golang/black-belt) could be a great way to correlate experience levels to book titles. 

A: The correlations are described in [Golang training at BIT](https://bitfieldconsulting.com/golang/bit).

**Comment by bitfield** [comment]
@cassamajor it's quite a significant rewrite, as you can probably tell from the code examples, so it's taking a while. I think it'll probably be out late summer / early autumn. Just to reiterate, though, if you [buy the book](https://bitfieldconsulting.com/books/tools) now, you'll get the updated edition for free, as will everyone who already owns the previous edition.

**Comment by bitfield** [comment]
@cassamajor the new edition is out now, if you're still interested 😄 

https://bitfieldconsulting.com/books/tools
